server/routes/reporting/monthly-utilisation/monthly-utilisation.controller.js

```diff
@@ -0,0 +1,122 @@
+    delete req.session.formFields;
+
+    return res.render('reporting/monthly-utilisation/prepare-select-month', {
+      months: generateSelectMonths(),
+      processUrl: app.namedRoutes.build('reports.prepare-monthly-utilisation.filter.post'),
+      cancelUrl: app.namedRoutes.build('reports.statistics.get'),
+      tmpBody,
+      errors: {
+        title: 'Please check your search',
+        count: typeof errors !== 'undefined' ? Object.keys(errors).length : 0,
+        items: errors,
+      },
+
+    });
+  };
+
+  module.exports.postPrepareMonthlyUtilisation = (app) => (req, res) => {
+    if (!req.body.selectMonth) {
+      req.session.formFields = req.body;
+      req.session.errors = makeManualError('selectMonth', 'Select a month to prepare report for');
+      return res.redirect(app.namedRoutes.build('reports.prepare-monthly-utilisation.filter.get'));
+    }
+
+    return res.redirect(app.namedRoutes.build('reports.prepare-monthly-utilisation.confirm.get', {
+      reportDate: req.body.selectMonth,
+    }));
+  };
+
+  module.exports.getCofirmPrepareMonthlyUtilisation = (app) => async(req, res) => {
+    const { reportDate } = req.params;
+
+    try {
+      const {response} = await courtDetailsDAO.get(app, req, req.session.authentication.locCode);
+      const courtName = capitalizeFully(response.english_court_name);
+
+      return res.render('reporting/monthly-utilisation/confirm-prepare', {
+        courtName,
+        month: dateFilter(reportDate, 'yyyy-MM-DD', 'MMMM YYYY'),
+        incompleteServiceReportUrl: app.namedRoutes.build('reports.incomplete-service.filter.get'),
+        continueUrl: app.namedRoutes.build('reports.prepare-monthly-utilisation.report.get', { filter: reportDate }),
+        cancelUrl: app.namedRoutes.build('reports.statistics.get'),
+      });
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  module.exports.getFilterMonthlyUtilisation = (app) => async(req, res) => {
+    const errors = req.session.errors || {};
+    const tmpBody = req.session.formFields || {};
+
+    delete req.session.errors;
+    delete req.session.formFields;
+
+    try {
+      const utilisationMonths = (await monthlyUtilisationReportsDAO.get(req, req.session.authentication.locCode)).data;
+
+      const months = utilisationMonths.split(',').filter(n => n).map((month) => {
+        return {
+          value: dateFilter(month, 'MMMM yyyy', 'yyyy-MM-DD'),
+          text: month,
+        };
+      });
+
+      return res.render('reporting/monthly-utilisation/view-select-month', {
+        months,
+        processUrl: app.namedRoutes.build('reports.view-monthly-utilisation.filter.post'),
+        cancelUrl: app.namedRoutes.build('reports.statistics.get'),
+        tmpBody,
+        errors: {
+          title: 'Please check your search',
+          count: typeof errors !== 'undefined' ? Object.keys(errors).length : 0,
+          items: errors,
+        },
+      });
+    } catch (e) {
+      console.log(e);
```

Comment:
same here, add a logger call and handle better.

Reply:
updated.